src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/ParameterNameCheck.java

```diff
@@ -105,16 +133,54 @@ public void setIgnoreOverridden(boolean ignoreOverridden) {
 
     @Override
     protected boolean mustCheckName(DetailAST ast) {
+        final Scope theScope = calculateScope(ast);
```

Comment:
we should not calculate scope if it is not defined by user.
Check need to work as before update on existing configurations.